pkg/controller/internal/harbor_commons.go

```diff
@@ -1,14 +1,20 @@
 package internal
 
 import (
+	"context"
 	"errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	h "github.com/mittwald/goharbor-client"
 	registriesv1alpha1 "github.com/mittwald/harbor-operator/pkg/apis/registries/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var ErrUserNotFound = errors.New("user not found")
 var ErrRegistryNotFound = errors.New("registry not found")
 var ErrReplicationNotFound = errors.New("replication not found")
+var ErrInstanceNotFound = errors.New("instance not found")
```

Comment:
Optional suggestion: Maybe use a type instead a value, and include the instance name?

```go
type ErrInstanceNotFound string

func (e ErrInstanceNotFound) Error() string {
    return fmt.Sprintf("instance '%s' not found", e)
}

// usage:
return ErrInstanceNotFound("some-instance-name")
```

Reply:
And make sure you explicitly convert `e` to string when passing it to `fmt.Sprintf`, because you might got problems otherwise:

```go
func (e ErrInstanceNotFound) Error() string {
    return fmt.Sprintf("instance '%s' not found", string(e))
}
```